src/ros_commo.py

```diff
@@ -308,30 +308,19 @@ def chat_perceived_text_cb(self, chat_heard):
 	# started, or stopped vocalizing.  This message might be published
 	# by either the TTS module itself, or by some external chatbot.
 	#
-	# Unit test by saying
-	#    rostopic pub --once chat_events std_msgs/String speechstart
-	#    rostopic pub --once chat_events std_msgs/String speechend
-	def chat_event_cb(self, chat_event):
-		print('chat_event, type ' + chat_event.data)
-		if chat_event.data == "start":
-			rospy.loginfo("webui starting speech")
+	def speech_event_cb(self, speech_event):
+		print('speech_event, type ' + speech_event.data)
+		if speech_event.data == "start":
+			rospy.loginfo("starting speech")
 			self.puta.vocalization_started()
-
-		elif chat_event.data == "stop":
+		elif speech_event.data == "stop":
+			rospy.loginfo("ending speech")
 			self.puta.vocalization_ended()
-			rospy.loginfo("webui ending speech")
-
-		elif chat_event.data == "listen_start":
-			self.puta.listening_started()
-			rospy.loginfo("webui ending speech")
-
-		elif chat_event.data == "listen_stop":
-			self.puta.listening_ended()
-			rospy.loginfo("webui ending speech")
-
-
+		elif speech_event.data.startswith("duration"):
```

Comment:
its not supposed to be a prediction its like elapsed time start to stop i guess. @wenwei-dev could clarify